yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -341,6 +342,17 @@ private[spark] class Client(
       env("SPARK_YARN_USER_ENV") = userEnvs
     }
 
+    // if spark.submit.pyArchives is in sparkConf, append pyArchives to PYTHONPATH
+    // that can be passed on to the ApplicationMaster and the executors.
+    if (sparkConf.contains("spark.submit.pyArchives")) {
+      var pythonPath = sparkConf.get("spark.submit.pyArchives")
+      if (env.contains("PYTHONPATH")) {
+        pythonPath = Seq(env.get("PYTHONPATH"), pythonPath).mkString(File.pathSeparator)
+      }
+      env("PYTHONPATH") = pythonPath
+      sparkConf.setExecutorEnv("PYTHONPATH", pythonPath)
```

Comment:
It feels there's something missing here.

If the archives are something that is not `local:`, shouldn't they be uploaded to HDFS and distributed using the cache? Otherwise there's no guarantee that the files exist remotely, right?